### PR TITLE
More Mediatek DSP updates

### PR DESCRIPTION
**Pull request by andyross**
Misc. work that was waiting for the earlier PR to merge.  Mostly this unbreaks SOF builds on these devices (SOF submission still to come) and fixes some 8196-specific edge cases.

### CODE REVIEW

**Comment by andyross** [comment]
Oh, and Intel SOF people (@kv2019i  @lyakh) will want to check the west sign changes as that impacts all platforms (but obviously should be a noop on existing devices).

File: tests/boards/mtk_adsp/src/main.c [hunk 1]
```diff
@@ -81,6 +79,8 @@ static void mbox_fn(const struct device *mbox, void *arg)
  */
 ZTEST(mtk_adsp, mbox)
 {
+	/* Different SOCs transmit the replies on different devices!  Just listen to both */
```
Comment (andyross) [inline tests/boards/mtk_adsp/src/main.c:82]: For those curious: yes, this happened without an ABI bump in the SOF kernel driver.  Not hard to accommodate both, even in real audio firmware, but.... ew.

File: soc/mediatek/mt8xxx/soc.c [hunk 2]
```diff
@@ -50,6 +50,27 @@ __asm__(".align 4\n\t"
 	"  movi  a1, " INIT_STACK "\n\t"
 	"  call4 c_boot\n\t");
 
+/* Unfortunately the SOF kernel loader doesn't understand the boot
+ * vector in the ELF/rimage file yet, so we still need a stub to get
```
Comment (kv2019i) [inline soc/mediatek/mt8xxx/soc.c:161]: Might be worth to file a https://github.com/thesofproject/sof/issues ticket and point to this.

**Review by kv2019i — APPROVED** [review]
Ack on the common parts in scripts/west_commands/ . Did a quick test build with SOF on Intel targets and seems all work as expected.

File: soc/mediatek/mt8xxx/soc.c (hunk 2 above)
Comment (andyross) [reply soc/mediatek/mt8xxx/soc.c:161]: I have a prototype fix already actually, but needs to wait for firmware submission that exploits an arbitrary boot vector.

**Comment by andyross** [comment]
This might be slipping off the leaderboard.  Ping for review.

**Review by nashif — APPROVED** [review]

**Comment by nashif** [comment]
@andyross  please submit a PR adding a new platform entry in the maintainer file with all the concerened filed etc so that PRs are correctly labelled and assigned.

**Pull request merged**